kafka/src/main/java/com/bazaarvoice/emodb/kafka/JsonPOJOSerde.java

```diff
@@ -11,20 +12,22 @@
 
 public class JsonPOJOSerde<T> implements Serde<T> {
 
-    private final ObjectMapper _mapper;
+    private static final ObjectMapper _mapper = new ObjectMapper();
     private final Class<T> _cls;
     private final TypeReference _typeReference;
 
+    static {
+        _mapper.registerModule(new JavaTimeModule());
```

Comment:
Considering that you are just converting the instant to epoch time in `MegabusRefResolver`, does it make sense to shed the java time module and just ship a `long` over the wire? Also, are we sure that this module won't affect the content of any documents in Emo that are also using this Serde?

Reply:
Well, since we're only using milliseconds resolution, I suppose it doesn't make any difference here, in terms of accuracy.

Is a `long` fewer bytes than a `String` here? Since a `String` is a `char[]`, and `Instant` will get serialized as something of a UTC date string (I believe), I would expect <30 characters or so. Let's say 32 bytes (in UTF-8). A long is a mere 8 bytes, so using `long` is likely a lot better for performance if you were to microbenchmark it.

It looks like `Instant#ofEpochMilli` can be used to create an `Instant` out of the `long`.

Good suggestion, I'll switch to `long` here.

As for not registering the JavaTimeModule, I feel like since the dependency is already available (notice I made no pom changes), it's already on the classpath and that makes me wonder if it's used elsewhere already.

Also, this JsonPOJOSerde is  only used by Megabus AFAIK, right? I think for any of `java.time`, if  you try to  serialize it, you'll get an exception. I don't know  if it is a good idea to install the module somewhat defensively,  since  I guess  things will be fine if  you first  deserialized something in more basic types (i.e., a date is just a String to Jackson unless you converted tthe String to a  `java.time` type). So if I switch  to `long` I can just uninstall this module and be fine. It shouldn't have any effect on serializiation of any other types because they have to  match the type  for the Jackson module during serde to  be invoked  AFAIK.

Reply:
Update: I checked and  we're actually _not_ using this  module elsewhere, but I believe we _should_ be. The dependency comes from `jackson-datatype-jsr310`, but instead of using `JavaTimeModule` we're using the (deprecated) `JSR310Module` in `CustomJsonObjectMapperFactory`.

Reply:
And one more update: I swapped to `long` and I gotta say, it doesn't feel ergonomic at all. I question whether or how much this affects storage/performance, and I don't feel like there's much risk in installing the Jackson `JavaTimeModule`. But if you feel strongly about it, I can go with long.

Reply:
Wow, you definitely did your due diligence on this one 😄 . I think leaving `JavaTimeModule` is fine, as the megabus always deserializes Emo documents from `Map<String, Object>`, which will shouldn't be affected.